pkg/sentry/socket/netlink/route/protocol.go

```diff
@@ -158,6 +161,81 @@ func (p *Protocol) dumpAddrs(ctx context.Context, hdr linux.NetlinkMessageHeader
 	return nil
 }
 
+// commonPrefixLen reports the length of the longest prefix.
+// This is a simplied version from Golang's src/net/addrselect.go.
+func commonPrefixLen(a, b []byte) (cpl int) {
+	for len(a) > 0 {
+		if a[0] == b[0] {
+			cpl += 8
+			a = a[1:]
+			b = b[1:]
+			continue
+		}
+		bits := 8
+		ab, bb := a[0], b[0]
+		for {
+			ab >>= 1
+			bb >>= 1
+			bits--
+			if ab == bb {
+				cpl += bits
+				return
+			}
+		}
+	}
+	return
+}
+
+// fillRoute returns the Route using LPM algorithm.
+// Refer to Linux's net/ipv4/route.c:rt_fill_info().
+func fillRoute(routes []inet.Route, addr []byte) (inet.Route, *syserr.Error) {
```

Comment:
This is a common function, which shall NOT be put into different stack implementations? And I don't see how this could be different in different stack implementations. No?

Reply:
Yeah, fair enough. I guess I'm just worried there might end up being differences in what netlink says and how the packets actually get routed. Though I suppose that could be treated as a bug in the underlying stack implementation.